### PR TITLE
Add Integration tests to Auditbeat CI pipeline

### PR DESCRIPTION
**Pull request by mjwolf**
## Proposed commit message

Refactor the CI integration test steps for the Auditbeat pipeline to use build matrices for Linux tests. By doing this, it will be easier to add additional VM types to the integration tests in the future.

With this, the integration tests will run on  Ubuntu 20.04, 22.04, 24.04 for both x86_64 and arm64, and RHEL 9 x86_64.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (mmat11) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:5]: which kernel version does this ship with?

File: .buildkite/auditbeat/scripts/integ-tests.sh
Comment (mmat11) [inline .buildkite/auditbeat/scripts/integ-tests.sh:11]: why is this needed?

**Comment by mmat11** [comment]
I guess these can be removed then https://github.com/elastic/beats/blob/main/auditbeat/Jenkinsfile.yml#L33-L50

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (mmat11) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:117]: nit: use `auditbeat: Integration Tests`, for consistency?

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (mmat11) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:142]: nit: use `auditbeat: Integration Tests`, for consistency?

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (mjwolf) [reply .buildkite/auditbeat/auditbeat-pipeline.yml]: It's using `5.15.0-1052-aws`

File: .buildkite/auditbeat/scripts/integ-tests.sh
Comment (mjwolf) [reply .buildkite/auditbeat/scripts/integ-tests.sh]: It's not needed, I've removed it. I took most of this script from the unit test script, but this isn't needed here

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (mjwolf) [reply .buildkite/auditbeat/auditbeat-pipeline.yml]: I've changed to this name

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (pkoutsovasilis) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:11]: maybe also add `family/core-sles-12` it is in the support matrix of `auditbeat`

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (pkoutsovasilis) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:4]: maybe also add `family/core-ubuntu-1804` it is in the support matrix of auditbeat

File: .buildkite/auditbeat/scripts/integ-tests.sh
Comment (oakrizan) [reply .buildkite/auditbeat/scripts/integ-tests.sh]: Buildkite-agent user has default umask=0002, meanwhile Jenkins has 0022. Without changing umask to 0022 tests are failing.

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (oakrizan) [reply .buildkite/auditbeat/auditbeat-pipeline.yml:12]: We are in process of migration beats from Jenkins to Buildkite, at the moment we are focusing on those agents, that were defined in Jenkinsfile and is available on Buildkite. Agent will be refined and added at the later stage, once all beats will be migrated to BK.

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (oakrizan) [reply .buildkite/auditbeat/auditbeat-pipeline.yml:11]: [Same](https://github.com/elastic/beats/pull/37892#discussion_r1481285263)

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (oakrizan) [reply .buildkite/auditbeat/auditbeat-pipeline.yml]: IMAGE_UBUNTU_X86_64: Linux bk-agent-prod-gcp-1707303793816829620 6.2.0-1019-gcp #21~22.04.1-Ubuntu SMP Thu Nov 16 18:18:34 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
IMAGE_UBUNTU_ARM_64: Linux i-095e3c2e7d81fa82d 5.15.0-1052-aws #57~20.04.1-Ubuntu SMP Mon Jan 15 17:04:12 UTC 2024 aarch64 aarch64 aarch64 GNU/Linux

**Comment by oakrizan** [comment]
> I guess these can be removed then https://github.com/elastic/beats/blob/main/auditbeat/Jenkinsfile.yml#L33-L50

Please keep the Jenkinsfiles in its' original state, since we still will need them

**Review by oakrizan — COMMENTED** [review]
All the improvements/new features have to be postponed until the migration is completed. Migration has a hard deadline and is ongoing.

**Comment by mjwolf** [comment]
> All the improvements/new features have to be postponed until the migration is completed. Migration has a hard deadline and is ongoing. cc: @dliappis @jlind23

I'll wait until the migration is completed before I merge this

**Comment by andrewkroh** [comment]
> I'll wait until the migration is completed before I merge this

I think buildkite the migration is mostly finished now. Is that correct, @dliappis?

So work on this could resume.

**Comment by dliappis** [comment]
> I think buildkite the migration is mostly finished now. Is that correct, @dliappis?

Correct.

**Comment by rowlandgeoff** [comment]
> either the VM image doesn't exist (Debian 10) or the image needs to be updated with additional tools to be able to run the tests (RHEL 8, SUSE 12). The images will need to be created/updated before they can be added to the integration tests.

Creating/updating these images will require planning and work on the Ingest Eng Prod team, and isn't something we can address during this PR. If you'd like to move forward with increasing your test coverage, I'd recommend pulling out the RHEL 8 and SLES 12 references until we can get to those. Assessing all of the beats and their test coverage relative to the compat matrix is something we were recently talking about, but it's going to take some time to assess and scope the work.

**Comment by mjwolf** [comment]
> > either the VM image doesn't exist (Debian 10) or the image needs to be updated with additional tools to be able to run the tests (RHEL 8, SUSE 12). The images will need to be created/updated before they can be added to the integration tests.
> 
> Creating/updating these images will require planning and work on the Ingest Eng Prod team, and isn't something we can address during this PR. If you'd like to move forward with increasing your test coverage, I'd recommend pulling out the RHEL 8 and SLES 12 references until we can get to those. Assessing all of the beats and their test coverage relative to the compat matrix is something we were recently talking about, but it's going to take some time to assess and scope the work.

Hi, @rowlandgeoff, thanks, I see now that a lot of the images in the `platform-ingest-beats` family don't exist.  I was planning to add them myself (at least the easy ones that can be based off the existing core images), but if your team is planning to do it, I can wait for it.

For this PR, I've removed the images that don't exist and updated the description with what I've done now

**Review by pkoutsovasilis — APPROVED** [review]
ty @mjwolf this LGTM! such a distros extension of running Auditbeat FIM integration tests is critical as we utilise kprobes and ebpf technologies which can be affected with kernel changes and updates. @rowlandgeoff eager to see this getting more extended to include more distros from the Elastic compatibility matrix 🙂

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (dliappis) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:155]: By using a matrix and having a static context we'll be facing the issue that all failures will be collapsed under the same GH status notification.

For example, right now, with passing tests, the notification looks like:

image

If we click the link it takes us to:

image

which represents both `Auditbeat Linux x86_64 Integration Tests -- family/platform-ingest-beats-ubuntu-2004` and `Auditbeat Linux x86_64 Integration Tests -- family/platform-ingest-beats-ubuntu-2204`

So if there is a failure, clicking on the link will take the user to the top of the matrix section, not the specific failed job. Looking at the [official BK docs](https://buildkite.com/docs/pipelines/build-matrix) it may not be possible to add `{{matrix.image}}` -- or some additional property lime `{{matrix.name}}` -- under `notify: ["github_commit_stats": {"context": "the message ... {{matrix.property}}"}]` (maybe worth testing to be 100% sure).

In any case, my main comment here is to raise awareness about the behavior and that with failures it won't be able to quickly see in the status section which specific image failed.

File: .buildkite/auditbeat/auditbeat-pipeline.yml
Comment (dliappis) [inline .buildkite/auditbeat/auditbeat-pipeline.yml:22]: nit: re: renaming `IMAGE_MACOS_ARM` to `IMAGE_MACOS_ARM64`, ideally we do this in a future PR where we change this as a convention for all the pipelines. For now, it might be ok to leave the name as is, given that there is no 32bit ARM architecture for macOS?

**Review by dliappis — COMMENTED** [review]
Looks good, thank you!

Left a comment to raise awareness on GH status notifications and that by using a matrix, failures will be collapsed under one entry. If you are ok with this I won't object, just wanted to make sure the behavior is clearly understood.

**Review by leehinman — APPROVED** [review]

**Comment by mjwolf** [comment]
I'd like to revive this PR.  I've added all the Linux 'platform-ingest-beats' VM images that currently exist and removed the variable name changes that weren't necessary. 

@leehinman, @dliappis, @pkoutsovasilis could you review this again, since it's changed since you last looked at it?

**Review by leehinman — APPROVED** [review]

**Comment by rowlandgeoff** [comment]
LGTM. @oakrizan can you please review? Any overlap with your current changes?

**Comment by oakrizan** [comment]
We deliberately were avoiding matrix usage. In addition to what Dimitrios said about matrix - if one of the matrix-based steps fails, then particular GH check will hang in "Pending" state.

**Review by oakrizan — APPROVED** [review]
lgtm

**Pull request merged**
